### PR TITLE
[EPMRPP-38820] Launch logs

### PR DESCRIPTION
**Pull request by ihar-kahadouski**

### CODE REVIEW

File: src/main/java/com/epam/ta/reportportal/core/log/impl/CreateLogHandlerImpl.java [hunk 1]
```diff
@@ -78,30 +82,35 @@
 	@Nonnull
 	//TODO check saving an attachment of the item of the project A in the project's B directory
 	public EntryCreatedAsyncRS createLog(@Nonnull SaveLogRQ request, MultipartFile file, ReportPortalUser.ProjectDetails projectDetails) {
+		Optional<TestItem> itemOptional = testItemRepository.findByUuid(request.getItemId());
+		Optional<Launch> launchOptional = launchRepository.findByUuid(request.getItemId());
```
Comment (pbortnik) [inline src/main/java/com/epam/ta/reportportal/core/log/impl/CreateLogHandlerImpl.java]: avoid excess database call

File: src/main/java/com/epam/ta/reportportal/core/log/impl/SaveLogBinaryDataTask.java [hunk 2]
```diff
@@ -123,7 +110,7 @@ public SaveLogBinaryDataTask withLaunchId(Long launchId) {
 	}
 
 	public SaveLogBinaryDataTask withItemId(Long itemId) {
-		Preconditions.checkNotNull(itemId, "Item id shouldn't be null");
+		//		Preconditions.checkNotNull(itemId, "Item id shouldn't be null");
```
Comment (pbortnik) [inline src/main/java/com/epam/ta/reportportal/core/log/impl/SaveLogBinaryDataTask.java]: ?

File: src/main/java/com/epam/ta/reportportal/ws/rabbit/LogReporterConsumer.java [hunk 3]
```diff
@@ -89,21 +95,30 @@ public void onLogCreate(@Payload DeserializablePair<SaveLogRQ, BinaryDataMetaInf
 		SaveLogRQ request = payload.getLeft();
 		BinaryDataMetaInfo metaInfo = payload.getRight();
 
-		Log log = new LogBuilder().addSaveLogRq(request).get();
-		TestItem testItem = testItemRepository.findByUuid(itemId)
-				.orElseThrow(() -> new ReportPortalException(TEST_ITEM_NOT_FOUND, itemId));
-		log.setTestItem(testItem);
+		Optional<TestItem> itemOptional = testItemRepository.findByUuid(request.getItemId());
```
Comment (pbortnik) [inline src/main/java/com/epam/ta/reportportal/ws/rabbit/LogReporterConsumer.java]: same with excess db call

File: src/main/java/com/epam/ta/reportportal/core/log/impl/CreateLogHandlerImpl.java [hunk 4]
```diff
@@ -78,30 +81,48 @@
...
+			saveBinaryData(file,
+					projectDetails.getProjectId(),
+					log.getId(),
+					testItemService.getEffectiveLaunch(item).getId(),
+					item.getItemId()
+			);
+		} else {
+			validate(request);
+			Launch launch = launchRepository.findByUuid(request.getItemId())
+					.orElseThrow(() -> new ReportPortalException(ErrorType.TEST_ITEM_NOT_FOUND, request.getItemId()));
```
Comment (pbortnik) [inline src/main/java/com/epam/ta/reportportal/core/log/impl/CreateLogHandlerImpl.java]: Launch not found

File: src/main/java/com/epam/ta/reportportal/core/log/impl/CreateLogHandlerImpl.java [hunk 5]
```diff
@@ -78,30 +81,48 @@
 	@Nonnull
 	//TODO check saving an attachment of the item of the project A in the project's B directory
 	public EntryCreatedAsyncRS createLog(@Nonnull SaveLogRQ request, MultipartFile file, ReportPortalUser.ProjectDetails projectDetails) {
+		Optional<TestItem> itemOptional = testItemRepository.findByUuid(request.getItemId());
+
+		Log log;
+		if (itemOptional.isPresent()) {
```
Comment (pbortnik) [inline src/main/java/com/epam/ta/reportportal/core/log/impl/CreateLogHandlerImpl.java]: separate into two separate privet methods

File: src/main/java/com/epam/ta/reportportal/core/log/impl/CreateLogHandlerImpl.java [hunk 6]
```diff
@@ -78,30 +81,48 @@
...
-					.withProjectId(projectDetails.getProjectId())
-					.withLaunchId(launchId)
-					.withItemId(testItem.getItemId())
-					.withLogId(log.getId()));
+					.withProjectId(projectId)
+					.withLogId(logId)
+					.withLaunchId(launchId);
 
-		}
+			if (Objects.isNull(itemId)) {
```
Comment (pbortnik) [inline src/main/java/com/epam/ta/reportportal/core/log/impl/CreateLogHandlerImpl.java]: is it expected ?

File: src/main/java/com/epam/ta/reportportal/core/log/impl/CreateLogHandlerImpl.java (hunk 1 above)
Comment (ihar-kahadouski) [reply src/main/java/com/epam/ta/reportportal/core/log/impl/CreateLogHandlerImpl.java]: fixed

File: src/main/java/com/epam/ta/reportportal/core/log/impl/SaveLogBinaryDataTask.java (hunk 2 above)
Comment (ihar-kahadouski) [reply src/main/java/com/epam/ta/reportportal/core/log/impl/SaveLogBinaryDataTask.java]: removed

File: src/main/java/com/epam/ta/reportportal/ws/rabbit/LogReporterConsumer.java (hunk 3 above)
Comment (ihar-kahadouski) [reply src/main/java/com/epam/ta/reportportal/ws/rabbit/LogReporterConsumer.java]: fixed

File: src/main/java/com/epam/ta/reportportal/core/log/impl/CreateLogHandlerImpl.java (hunk 4 above)
Comment (ihar-kahadouski) [reply src/main/java/com/epam/ta/reportportal/core/log/impl/CreateLogHandlerImpl.java]: fixed

File: src/main/java/com/epam/ta/reportportal/core/log/impl/CreateLogHandlerImpl.java (hunk 5 above)
Comment (ihar-kahadouski) [reply src/main/java/com/epam/ta/reportportal/core/log/impl/CreateLogHandlerImpl.java]: done

File: src/main/java/com/epam/ta/reportportal/core/log/impl/CreateLogHandlerImpl.java (hunk 6 above)
Comment (ihar-kahadouski) [reply src/main/java/com/epam/ta/reportportal/core/log/impl/CreateLogHandlerImpl.java]: fixed

**Pull request merged**